include/mvdtool/bits/mvd3_misc.hpp

```diff
@@ -185,9 +187,10 @@ inline Rotations MVD3File::getRotations(const Range& range) const {
     HighFive::DataSet set = _hdf5_file.getDataSet(did_cells_rotations);
     if (!range.is_empty()) {
```

Comment:
My only remark is that `is_empty` now is misleading. Maybe we could drop it and have an expressive
`range == range::all()`. (Maybe all can even have a `constexpr all` object ?

Reply:
I agree that the name is not representative.. Maybe simply renaming it to `is_until_end` is sufficient.
From my point of view, `is_empty` checks if ` count == 0`, meaning that even in case that the range is instantiated like `Range(offset, 0)` it should return `true`, since this would mean all the elements from `offset` until the `end`.
What do you think about that?

Reply:
right, the conditions would have to change. At the moment I find not so cool that in both il/else parts we have to compute the range.
What I have in mind is something like python slices. Apparently now that's all hidden but, anyway, check https://docs.python.org/3/c-api/slice.html#c.PySlice_AdjustIndices
So instead of checking for a particular case when `count == 0` adjust in some way, we could have a nicer OO way and if not range::all, do `count = range.adjust_indices(seq_length)`

Reply:
That's a very good point! :+1:
I am working on it

Reply:
I refactored this a bit. Let me know what you think :slightly_smiling_face: